app/routes/campaign.js

```diff
@@ -17,8 +17,11 @@ module.exports = function (router) {
     router.post('/campaign/create', (request, response) => {
         if(request.user) {
             campaignPromise = campaignController.saveCampaign(request);
-            campaignPromise.then(function(campaign) {
-                response.send(campaign._id);
+            campaignPromise.then(function() {
+                return campaignController.findCampaignByTitleAndUser(request.body.title, request.user.id);
+            })
+            .then(function(result) {
+                response.send(result[0]._id);
```

Comment:
Yea I ran it just with "save" and got an empty return (I saw no shortid on the success page). I'm a little confused by this since I thought when I had tried this before, it was behaving differently, But yea to get consistent behavior I added it back.

It's possible that adding an "exec" call on save may fix this issue.